webservices/gotchas/javascript.md

```diff
@@ -0,0 +1,114 @@
+## Gotchas with common javascript functions
+
+###PUT, PATCH and DELETE ajax requests 
+
+####Requests that manipulate data need CSRF tokens
+
+CSRF stands for `Cross Site Request Forgery`. Effectively what a CSRF requirement is attempting to prevent is the ability for another site to trick a user into clicking a link that may cause malicious behavior such as deleting your account, or transfer money somewhere.
+
+A detailed account can be found [here].
+[here]: http://stackoverflow.com/questions/941594/understanding-the-rails-authenticity-token
+
+This is a sample implementation of a submission through and AJAX call. Using the Rails built in forms or [simple form for] will do this automatically. However sometimes this is not ideal if the elements you are selecting are loaded via another AJAX call.
+
+[simple form for]: https://github.com/plataformatec/simple_form
+
+**Form to Delete**
+Create a Rails button and use this special `token_tag` generator to create a CSRF token on the page.
+```
+<h3>Delete endorsement for <%= user.name %></h3>
+<p class="modal-content">Are you sure you wish to delete this endorsement? This action cannot be undone. </p>
+<%= token_tag nil %>
+<%= submit_tag 'Confirm Delete', type: 'button', class: "delete-button" %>
+```
+
+**Javascript to Send AJAX Call**
+grab the id's you need from elements on the page in order to build your url. Then add then token as an `authenticity_token` in the `data` portion of the AJAX call.
+
+```
```

Comment:
might want to specify the language for nice markdown syntax highlighting: `/`/`/`javascript`